Add support for toggle buttons via RC channels

cmic0 opened the pull request:
**Describe problem solved by this pull request**
This PR introduces a new configuration option which allows users to setup Toggle Buttons for flight mode switches, over the RC signals (e.g S.BUS).

An example application is for example with the Herelink controller, where the buttons can be assigned to specific RC channels which are then sent over S.BUS.

I also added a 100ms hysteresis before flagging that the signal has been regained, as I could experience some glitches on the S.BUS channels when the controller was rebooted. 

**Test data / coverage**
Tested with a herelink controller

dagar commented:
I had a similar version of this where you could optionally treat any RC switch like a button, then the button handling was centralized in commander where each could be configured for a particular command like arm/disarm, mode changes, and other vehicle commands.

The benefit was the configuration was now unified for both RC buttons and Mavlink MANUAL_CONTROL buttons, and it was a bit easier to start adding support for things like light control or other random vehicle_commands.

Is that worth discussing briefly? I only bring this up to possibly avoid churn later, otherwise this PR looks fine to me if you want to get it in now.

cmic0 commented:
@bkueng @dagar @MaEtUgR i am resurrecting this PR with the hope that we can get it in before https://github.com/PX4/PX4-Autopilot/pull/17404 (i think it's easier to handle the small diff in there)

dagar commented on src/modules/rc_update/params.c:
```diff
@@ -1702,6 +1702,41 @@ PARAM_DEFINE_INT32(RC_MAP_STAB_SW, 0);
+ * parameters.
+ * The functionality can be used only if RC_MAP_FLTMODE is disabled.
+ *
+ * The maximum number of available slots is 6.
```
Missing group.

cmic0 replied:
fixed

dagar commented:
At the moment I'm not quite sure how to reconcile this with https://github.com/PX4/PX4-Autopilot/pull/17404, but I'm not opposed to bringing this in first.

I think we'll want much more flexible button handling regardless of source, but we'll cross that bridge later.

cmic0 commented:
@dagar addressed the review comment, thanks for reviewing. This is good to go IMO.

> I think we'll want much more flexible button handling regardless of source, but we'll cross that bridge later.

For general context: ideally you would centralize all the various sources that are responsible for flight mode changes (e.g the `VEHICLE_CMD_DO_SET_MODE` commands that are directly handled in commander) in one single place?

cmic0 commented:
@dagar / @bkueng CI is failing to compile for px4_fmu-v2 due to flash overflow :-1: , anything you can do there?

MaEtUgR commented:
I rebased again on latest master and added my suggestions as a commit. Sry @bkueng I didn't finish yesterday. Could you check just that commit?

cmic0 reviewed (commented):
Thanks @MaEtUgR for improving code readability, looks good!
